### PR TITLE
connectivity assert col_names == row_names (#162)

### PR DESCRIPTION
**Pull request by AhmetNSimsek**

### CODE REVIEW

**Comment by xgui3783** [comment]
rather than add strict check in siibra, I think it's far more useful to add it as e2e test, so that when/if new datasets are added, violating this rule, it will be caught by CI, and not at runtime

**Comment by AhmetNSimsek** [comment]
If someones uses their own data, then this check is still necessary imo.

**Comment by xgui3783** [comment]
> If someone uses their own data, then this check is still necessary IMO. Can still add the tests as an extra but as long as get_matrix is tested, this will be included.

User will need to call `get_matrix` on every subject to see the error (unless I am mistaken?). When using custom data, they could run ci/cd to validate their data.

Not to mention, adding assertion at runtime taxes every call to `get_matrix`, whereas in reality, it really just need to be done once.

File: e2e/features/connectivity/test_connectivity.py [hunk 1]
```diff
@@ -22,9 +22,13 @@ def test_connectivity_get_matrix(f: RegionalConnectivity):
     assert isinstance(f, RegionalConnectivity)
     assert len(f.subjects) > 0
     assert all(isinstance(subject, str) for subject in f.subjects)
+    matrix_df = f.get_matrix(subject=FileNotFoundError)
+    assert all(matrix_df.index[i] == r for i, r in enumerate(matrix_df.columns))
```
Comment (xgui3783) [inline e2e/features/connectivity/test_connectivity.py:26]: subject=filenotfound?

Comment (AhmetNSimsek) [reply e2e/features/connectivity/test_connectivity.py:26]: This is for testing the averaged matrix which requires `subject=None`.
We've recently made sure that this is the way instead using `subject='mean'`

**Comment by AhmetNSimsek** [comment]
Note: The failing tests are due to bigbrain intensity and is not relevant.

File: e2e/features/connectivity/test_connectivity.py (hunk 1 above)
Comment (xgui3783) [reply e2e/features/connectivity/test_connectivity.py:26]: shouldn't it be `f.get_matrix()` or `f.get_matrix(None)` then?

Comment (AhmetNSimsek) [reply e2e/features/connectivity/test_connectivity.py:26]: Noth works at the end but eventually, it just takes `subject=None`. But perhaps better to use `f.get_matrix()` as the default behaviour might change later.

Comment (xgui3783) [reply e2e/features/connectivity/test_connectivity.py:26]: but you used `f.get_matrix(subject=FileNotFoundError)`

am I missing something?

File: e2e/features/connectivity/test_connectivity.py [hunk 2]
```diff
@@ -22,9 +22,13 @@ def test_connectivity_get_matrix(f: RegionalConnectivity):
     assert isinstance(f, RegionalConnectivity)
     assert len(f.subjects) > 0
     assert all(isinstance(subject, str) for subject in f.subjects)
+    matrix_df = f.get_matrix(subject=FileNotFoundError)
```
Comment (xgui3783) [inline e2e/features/connectivity/test_connectivity.py:25]:
```suggestion
    matrix_df = f.get_matrix()
```

File: e2e/features/connectivity/test_connectivity.py (hunk 1 above)
Comment (xgui3783) [reply e2e/features/connectivity/test_connectivity.py:26]: see https://github.com/FZJ-INM1-BDA/siibra-python/pull/347/files#r1172494587

Comment (AhmetNSimsek) [reply e2e/features/connectivity/test_connectivity.py:26]: Oh I thought I did `subject=None`. That is very weird...

Comment (AhmetNSimsek) [reply e2e/features/connectivity/test_connectivity.py]: Corrected with [b76161b](https://github.com/FZJ-INM1-BDA/siibra-python/pull/347/commits/b76161b5a5cc3831f2a27b4fa2e8cefebe717240)

**Comment by AhmetNSimsek** [comment]
(The example tests failed due to network error in Allen inst.)

**Pull request merged**